src/main/kotlin/me/ezra_home/retail_software_solution/model/entity/FruitEntity.kt

```diff
@@ -0,0 +1,51 @@
+package me.ezra_home.retail_software_solution.model.entity
+
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.Id
+import jakarta.persistence.Table
+import jakarta.validation.constraints.NotNull
+import jakarta.validation.constraints.Size
+import org.hibernate.annotations.ColumnDefault
+import java.math.BigDecimal
+import java.time.OffsetDateTime
+import java.util.*
+
+@Entity(name = "FruitEntity")
+@Table(name = "fruit", schema = "public")
+open class FruitEntity {
+    @Id
+    @ColumnDefault("gen_random_uuid()")
+    @Column(name = "id", nullable = false)
```

Comment:
- There is no harm is adding `nullable=false`. However `@Id` already implies that it cant be nullable. 
- You can mark `updateable=false` though to prevent accidental edits being made on the id
- That being said, there is a BaseEntity Class that you can/should extend to avoid defining the id field